client/src/controllers/BulkController.ts

```diff
@@ -0,0 +1,66 @@
+import { Controller } from '@hotwired/stimulus';
+/**
+ * Adds the ability to collectively toggle a set of (non-disabled) checkboxes.
+ *
+ * @example
+ * <div data-controller="w-bulk">
+ *   <input type="checkbox" data-action="w-bulk#toggleAll" data-w-bulk-target="all">
+ *   <div>
+ *     <input type="checkbox" data-action="w-bulk#change" data-w-bulk-target="item" disabled>
+ *     <input type="checkbox" data-action="w-bulk#change" data-w-bulk-target="item">
+ *     <input type="checkbox" data-action="w-bulk#change" data-w-bulk-target="item">
+ *   </div>
+ *   <button data-action="w-bulk#toggleAll" data-w-bulk-force-param="false">Clear all</button>
+ *   <button data-action="w-bulk#toggleAll" data-w-bulk-force-param="true">Select all</button>
+ * </div>
+ */
+export class BulkController extends Controller<HTMLElement> {
+  static targets = ['all', 'item'];
+
+  /** All select-all checkbox targets */
+  declare readonly allTargets: HTMLInputElement[];
+
+  /** All item checkbox targets. */
```

Comment:
```suggestion
  /** All item checkbox targets */
```

minor nitpick, let's be consistent and not add periods, the other JSDoc items do not have this. Sorry if this was from my suggestion.